### PR TITLE
[CI][Bugfix] Align the LTX-2 FP8 quality gate with the default recipe

### LINKED ISSUE #5733: [Bug]: Nightly / CI failed Occasional- tests/diffusion/quantization/test_quantization_quality.py::test_quantization_quality[fp8_ltx2], AssertionError: LPIPS 0.1102 exceeds threshold 0.1 for fp8 on Lightricks/LTX-2
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
ci env
```

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
(not found in Buildkite step log)
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
900a7f0813d0
```
</details>

### 🐛 Describe the bug

**Buildkite build:** https://buildkite.com/vllm/vllm-omni/builds/13108
**Buildkite step:** [:full_moon: Diffusion · Quantization Test with H100](https://buildkite.com/vllm/vllm-omni/builds/13108#019fc8cb-5601-4bea-be13-956f5a3aa2e4)

**Failure kind:** pytest FAILED
**Test node:** `tests/diffusion/quantization/test_quantization_quality.py::test_quantization_quality[fp8_ltx2]`

**Log reason:**
AssertionError: LPIPS 0.1102 exceeds threshold 0.1 for fp8 on Lightricks/LTX-2

**Analysis:**
Category: assertion failure. Expected vs actual mismatch — check test logic, edge cases, and data setup.

**Error log excerpt:**

```text
config = QualityTestConfig(id='fp8_ltx2', task='t2v', prompt='A serene lakeside sunrise with mist over the water', max_lpips=0....7483, 0.6111654043197632, 0.4709382653236389, 0.30347853899002075, 0.10000002384185791, 0.0], enable_cpu_offload=False)



    @pytest.mark.full_model

    @pytest.mark.diffusion

    @pytest.mark.parametrize(

        "config",

        [_quality_param(c) for c in _all_quality_configs()],

    )

    def test_quantization_quality(config: QualityTestConfig):

        """Validate that quantized output stays within LPIPS threshold of BF16."""

        from vllm_omni.entrypoints.omni import Omni

    

        generate_fn = _generate_video if config.task == "t2v" else _generate_image

    

        # --- BF16 baseline ---

        bl_kwargs: dict = {"model": config.baseline_ref()}

        if config.enable_cpu_offload:

            bl_kwargs["enable_cpu_offload"] = True

        omni_bl = Omni(**bl_kwargs)

        baseline_out, bl_mem = generate_fn(omni_bl, config)

        omni_bl.shutdown()

        del omni_bl

        _free_gpu_memory()

        _maybe_save_output(_OUTPUT_DIR, config, "baseline", baseline_out)

    

        # --- Quantized ---

        quantization = config.quantization_ref()

        qt_kwargs: dict = {"model": config.quantized_ref()}

        if config.enable_cpu_offload:

            qt_kwargs["enable_cpu_offload"] = True

        if quantization is None:

            omni_qt = Omni(**qt_kwargs)

        else:

            omni_qt = Omni(**qt_kwargs, quantization_config=quantization)

        quant_out, qt_mem = generate_fn(omni_qt, config)

        omni_qt.shutdown()

        del omni_qt

        _free_gpu_memory()

        _maybe_save_output(_OUTPUT_DIR, config, "quantized", quant_out)

    

        # --- Similarity metrics ---

        lpips_score = _compute_lpips(baseline_out, quant_out, config.task)

        psnr_score, mae_score = _compute_psnr_and_mae(baseline_out, quant_out, config.task)

>       assert lpips_score <= config.max_lpips, (

            f"LPIPS {lpips_score:.4f} exceeds threshold {config.max_lpips} "

            f"for {config.quantization_ref() or 'pre-quantized checkpoint'} on {config.quantized_ref()}"

        )

E       AssertionError: LPIPS 0.1102 exceeds threshold 0.1 for fp8 on Lightricks/LTX-2

E       assert 0.11023492634296417 <= 0.1

E        +  where 0.1 = QualityTestConfig(id='fp8_ltx2', task='t2v', prompt='A serene lakeside sunrise with mist over the water', max_lpips=0.1, model='Lightricks/LTX-2', quantization='fp8', baseline_model=None, quantized_model=None, height=25

...(truncated for GitHub URL length; see HTML report for full log)
```

---
*Generated from a nightly HTML report. Redact secrets before submitting; complete the checkboxes on GitHub.*

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by mglyn**
## Purpose

Fixes [#5733](https://github.com/vllm-project/vllm-omni/issues/5733).

The old `fp8_ltx2` case pinned a pre-#5148 custom sigma trajectory. The current LTX pipeline intentionally uses the official multimodal guidance recipe, so this mixed old scheduling with new guidance and exceeded the historical LPIPS threshold.

The gate now measures the supported eager default recipe:

- Lightricks/LTX-2 online FP8 versus BF16;
- `512x384`, 73 frames, 10 steps, seed 42;
- no request sigma or guidance override;
- recipe-default negative conditioning;
- `enforce_eager=True` for both runs, so compile behavior is not part of a
  quantization-quality assertion.

The LPIPS threshold follows the standard video quality gate at `0.20`; this PR does not change quantization implementation.

## Test Result

```bash
python -m pytest -q \
  tests/diffusion/quantization/test_quantization_quality.py \
  -m '' -k 'test_quantization_quality and fp8_ltx2' -s
```

| Metric | Result | Acceptance |
|---|---:|---:|
| LPIPS | 0.0418 | <= 0.20 |
| PSNR | 33.0941 dB | Reported |
| MAE | 0.017016 | Reported |

The small configuration tests and changed-file pre-commit hooks also pass.

### CODE REVIEW

File: tests/diffusion/quantization/test_quantization_quality.py [hunk 1]
```diff
@@ -525,6 +536,8 @@ def test_quantization_quality(config: QualityTestConfig):
     bl_kwargs: dict = {"model": config.baseline_ref(), "enforce_eager": True}
     if config.enable_cpu_offload:
         bl_kwargs["enable_cpu_offload"] = True
+    if config.enforce_eager:
```
Comment (Gaohan123) [inline tests/diffusion/quantization/test_quantization_quality.py]: `bl_kwargs` is already initialized with `enforce_eager=True`, and the same applies to `qt_kwargs`. Therefore, these new conditional assignments—and effectively the new `config.enforce_eager` field—cannot change execution behavior. Please either remove the field, both branches, and the corresponding assertion, or remove the hardcoded values from the initial dictionaries so the configuration actually controls eager mode.

File: tests/diffusion/quantization/test_quantization_quality.py [hunk 2]
```diff
@@ -490,6 +488,19 @@ def generate(self, _prompt, sampling):
     assert captured.sampling.sigmas == [1.0, 0.5]
 
 
+@pytest.mark.core_model
+@pytest.mark.cpu
+def test_ltx_quality_gate_uses_official_eager_defaults():
```
Comment (Gaohan123) [inline tests/diffusion/quantization/test_quantization_quality.py]: This test only checks `QualityTestConfig` attributes and never exercises `_generate_video`, so it would still pass if the generation path accidentally started forwarding `negative_prompt`, `sigmas`, or `guidance_scale` overrides again. Please extend it using a capturing `DummyOmni`, similar to `test_generate_video_forwards_sigmas`, and assert that `negative_prompt` is absent from the prompt while `sampling.sigmas` and `sampling.guidance_scale` are `None`.

File: tests/diffusion/quantization/test_quantization_quality.py (hunk 1 above)
Comment (mglyn) [reply tests/diffusion/quantization/test_quantization_quality.py]: Addressed in 71ccf6c4. Removed QualityTestConfig.enforce_eager, both redundant conditionals, and the corresponding assertion. The quality gate keeps enforce_eager=True hardcoded for both the BF16 and FP8 runs, so eager behavior remains explicit and consistent.

File: tests/diffusion/quantization/test_quantization_quality.py (hunk 2 above)
Comment (mglyn) [reply tests/diffusion/quantization/test_quantization_quality.py]: Addressed in 71ccf6c4. The test now calls _generate_video with a capturing DummyOmni and asserts that negative_prompt is omitted and sampling.sigmas / sampling.guidance_scale are None. The existing sigma-forwarding mock was also updated to return OmniRequestOutput. Validation: pre-commit passed; the local LTX-2 FP8 quality case passed on a canhazgpu-scheduled GPU using /data/models (LPIPS 0.0418 <= 0.20, PSNR 33.0941 dB, MAE 0.017016).

**Comment by mglyn** [comment]
This looks like a valid small fix for the existing LTX-2 FP8 gate. Since that gate is currently skipped under #6245 and LTX-2.5 is now supported, could we first confirm whether the legacy LTX-2 FP8 gate is still intended to be maintained? If so, this PR seems reasonable; otherwise, a current LTX-2.5 FP8 quality case may be more useful than retaining a skipped legacy gate.

**Comment by Gaohan123** [comment]
> This looks like a valid small fix for the existing LTX-2 FP8 gate. Since that gate is currently skipped under #6245 and LTX-2.5 is now supported, could we first confirm whether the legacy LTX-2 FP8 gate is still intended to be maintained? If so, this PR seems reasonable; otherwise, a current LTX-2.5 FP8 quality case may be more useful than retaining a skipped legacy gate.

I think we should maintain now. As the downstream RL training like verl-omni still relies on it

**Review by Gaohan123 — COMMENTED** [review]
Please recover the test case. Thanks

**Comment by mglyn** [comment]
I did not remove any test case, you mean recover to "256 x 256", old fixed smaller sigmas? @Gaohan123

File: tests/diffusion/quantization/test_quantization_quality.py
Comment (Gaohan123) [inline tests/diffusion/quantization/test_quantization_quality.py]: @mglyn This one. You should recover the case which is skipped previously

Comment (mglyn) [reply tests/diffusion/quantization/test_quantization_quality.py]: Recovered in c13729fac by removing the fp8_ltx2 skip. The full quality case passes locally (LPIPS 0.0418 <= 0.20). Thanks!

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks
